framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -1523,6 +1522,39 @@ private void handleExpr(AnnotationBuilder builder, String name, Expression expr)
                                 + " and expected: "
                                 + expected);
             }
+        } else if (expr instanceof ClassExpr) {
+            ClassExpr classExpr = (ClassExpr) expr;
+            ExecutableElement var = builder.findElement(name);
+            TypeMirror expected = var.getReturnType();
+            if (expected.getKind() == TypeKind.DECLARED) {
+                String className = classExpr.getType().toString();
+                Class<?> clazz;
+                try {
+                    // If it is a simply-named class literal then fully-qualified class
```

Comment:
Where is a check whether it is a simply-named class literal? Don't you always go through all imports?

Reply:
Not sure how to implement it. I thought on checking that class name contains ".", but it won't work with Outer.Inner.class.
If you could think of something please suggest?

Reply:
Found a way. Check first letter case.

Reply:
That is just a heuristic.  It may fail in obscure cases (when users don't obey Java naming standards, but that does happen).  So, I wouldn't use the heuristic.

Reply:
I agree with Mike that such a heuristic is not a good idea.

Have you looked at method `getSupportedAnnotations` https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/stub/StubParser.java#L238
and helpers like `getImportableMembers`?
Instead of going through the `ImportDeclaration`s here, I think it would be nicer to extend that existing logic to keep around all imported types. Only such types can then later be used in `.class` literals. We already have to go through the imports then and it seems wasteful to go through all of that work for every class literal.
The logic here would then first go to this imported-types map and see whether the class literal string is contained there. If it is contained, you use the expansion. If it's not contained, you use the type string. Then you do the `Class.forName`. If the type is a fully-qualified class name, it will not be contained in the imports-map, but the `Class.forName` will still succeed.
Does this make sense?

Reply:
Thank you! Working on it.